x-pack/legacy/plugins/infra/public/components/waffle/node_context_menu.tsx

```diff
@@ -43,83 +48,116 @@ export const NodeContextMenu = ({
       nodeType,
       nodeId: node.id,
       from: nodeDetailFrom,
       to: currentTime,
     }),
+    isDisabled: !showDetail,
   };
 
-  const apmTracesMenuItem = {
-    name: i18n.translate('xpack.infra.nodeContextMenu.viewAPMTraces', {
-      defaultMessage: 'View APM traces',
+  const apmTracesMenuItem: EuiListGroupItemProps = {
+    label: i18n.translate('xpack.infra.nodeContextMenu.viewAPMTraces', {
+      defaultMessage: '{inventoryName} APM traces',
+      values: { inventoryName: inventoryModel.singularDisplayName },
     }),
     href: `../app/apm#/traces?_g=()&kuery=${apmField}:"${node.id}"`,
     'data-test-subj': 'viewApmTracesContextMenuItem',
+    isDisabled: !showAPMTraceLink,
   };
 
-  const uptimeMenuItem = {
-    name: i18n.translate('xpack.infra.nodeContextMenu.viewUptimeLink', {
-      defaultMessage: 'View in Uptime',
+  const uptimeMenuItem: EuiListGroupItemProps = {
+    label: i18n.translate('xpack.infra.nodeContextMenu.viewUptimeLink', {
+      defaultMessage: '{inventoryName} in Uptime',
+      values: { inventoryName: inventoryModel.singularDisplayName },
     }),
     href: createUptimeLink(options, nodeType, node),
+    isDisabled: !showUptimeLink,
   };
 
-  const showDetail = inventoryModel.crosslinkSupport.details;
-  const showLogsLink =
-    inventoryModel.crosslinkSupport.logs && node.id && uiCapabilities?.logs?.show;
-  const showAPMTraceLink =
-    inventoryModel.crosslinkSupport.apm && uiCapabilities?.apm && uiCapabilities?.apm.show;
-  const showUptimeLink =
-    inventoryModel.crosslinkSupport.uptime && (['pod', 'container'].includes(nodeType) || node.ip);
-
-  const items = [
-    ...(showLogsLink ? [nodeLogsMenuItem] : []),
-    ...(showDetail ? [nodeDetailMenuItem] : []),
-    ...(showAPMTraceLink ? [apmTracesMenuItem] : []),
-    ...(showUptimeLink ? [uptimeMenuItem] : []),
-  ];
-  const panels: EuiContextMenuPanelDescriptor[] = [{ id: 0, title: '', items }];
-
-  // If there is nothing to show then we need to return the child as is
-  if (items.length === 0) {
-    return <>{children}</>;
-  }
-
   return (
-    <EuiPopover
+    <ActionMenu
       closePopover={closePopover}
       id={`${node.pathId}-popover`}
       isOpen={isPopoverOpen}
       button={children}
-      panelPaddingSize="none"
       anchorPosition={popoverPosition}
     >
-      <EuiContextMenu initialPanelId={0} panels={panels} data-test-subj="nodeContextMenu" />
-    </EuiPopover>
+      <Section>
+        <SectionTitle>
+          <FormattedMessage
+            id={'xpack.infra.nodeContextMenu.title'}
+            defaultMessage={'{inventoryName} details'}
+            values={{ inventoryName: inventoryModel.singularDisplayName }}
+          />
+        </SectionTitle>
+        <SectionSubtitle>
+          <FormattedMessage
+            id={'xpack.infra.nodeContextMenu.description'}
+            defaultMessage={
+              'View logs, metrics and traces of this {inventoryName} to get further details'
```

Comment:
Some `inventoryName`s like `Host` look odd when used in the middle of a sentence, because they are capitalized. Not sure how whether that's actually a problem or how to resolve it, though.

Reply:
yeah, per Katrin, I actually think this needs to be replaced with ip address anyway.